### PR TITLE
feat(service-catalog): Create ArgoCD app

### PR DESCRIPTION
**Pull request by SamoKopecky**
Part of #2036 

## Changes
- Create ArgoCD app for service-catalog
- Create prostgres-db bundle required by the service-catalog

Right now blocked by [#2](https://github.com/operate-first/service-catalog/pull/2).

### CODE REVIEW

**Review by tumido — CHANGES_REQUESTED** [review]
it's on a good path, few improvements are needed to make this mergeable :slightly_smiling_face:

**Comment by SamoKopecky** [comment]
Thanks for the suggestions! Ready for review again.

File: argocd/overlays/moc-infra/projects/operate-first.yaml [hunk 1]
```diff
@@ -75,3 +75,7 @@ spec:
       kind: ThanosRuler
     - group: logging.openshift.io
       kind: ClusterLogForwarder
+    - group: operators.coreos.com
+      kind: OperatorGroup
+    - group: operators.coreos.com
+      kind: Susbscription
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/projects/operate-first.yaml:81]: the subscription and operatorgroup are not being deployed by the app you are creating here, but rather the `cluster-resources-smaug`, which already has the permissions to deploy these, so you can revert the changes to this project.

Comment (SamoKopecky) [reply argocd/overlays/moc-infra/projects/operate-first.yaml]: Thanks! Should be fixed now.

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/operate-first/service-catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  name: service-catalog
+spec:
+  project: operate-first
+  source:
+    repoURL: https://github.com/operate-first/service-catalog.git
+    targetRevision: main
+    path: manifests/overlays/prod
+  destination:
+    name: smaug
+    namespace: service-catalog
```
[the remark this answers is not in the document]
Comment (SamoKopecky) [reply argocd/overlays/moc-infra/applications/envs/moc/smaug/operate-first/service-catalog.yaml:13]: Really sorry about that, I should have looked into it more before pushing. It should be added correctly now.

[the remark this answers is not in the document]
Comment (HumairAK) [reply argocd/overlays/moc-infra/applications/envs/moc/smaug/operate-first/service-catalog.yaml:13]: yeah without v1beta1, whoops
